pinot-core/src/main/java/org/apache/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -95,25 +94,19 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
 
     // Leaf node
     String column = filterQueryTree.getColumn();
-    ColumnMetadata metadata = columnMetadataMap.get(column);
-    if (metadata == null) {
-      return false;
+    ColumnMetadata columnMetadata = columnMetadataMap.get(column);
+    // NOTE: should have already been pruned in DataSchemaSegmentPruner
```

Comment:
This will change the behavior. I have checked that `DataSchemaSegmentPruner` is checking if all columns from the query exists in the query request. Checking column is enough for checking column metadata existence?

Reply:
That is correct. But if we use this pruner alone (not with `DataSchemaSegmentPruner`), we should check whether ColumnMetadata exists.